code/modules/lighting/light_atom_set.dm

```diff
@@ -0,0 +1,113 @@
+// Destroys and removes a light
+/atom/proc/kill_light()
+	if(light_obj)
+		qdel(light_obj)
+		light_obj = null
+	if (shadow_obj)
+		qdel(shadow_obj)
+		shadow_obj = null
+	if (smooth_light_obj)
```

Comment:
shouldn't it be qdel(smooth_light_obj)

Reply:
It should be and `QDEL_NULL()` is one of TG's actually reasonable macros and we should steal it

Reply:
we just did. it's time